core/src/main/java/com/datastax/oss/driver/internal/core/data/DefaultTupleValue.java

```diff
@@ -93,6 +98,78 @@ private void readObject(ObjectInputStream stream) throws InvalidObjectException
     throw new InvalidObjectException("Proxy required");
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (!(o instanceof TupleValue)) {
+      return false;
+    }
+    DefaultTupleValue that = (DefaultTupleValue) o;
```

Comment:
Oops, I had missed this: it should cast to `TupleValue` here, and use the interface methods to access the fields. Otherwise it might throw if the other object is a custom implementation.